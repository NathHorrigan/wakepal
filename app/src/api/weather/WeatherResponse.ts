// Enum that maps the weather description to an icon supported by the UI
export enum WeatherIcon {
  Sunny = 'Sunny',
  PartlySunny = 'PartlySunny',
  Cloudy = 'Cloudy',
  Gloomy = 'Gloomy',
  Rainy = 'Rainy',
  SunnySpells = 'SunnySpells',
  Lightning = 'Lightning',
  Snow = 'Snow',
  Fog = 'Fog',
  Moon = 'Moon',
}

// Response that must be returned by any Weather Api controller
export class WeatherResponse {
  public createdAt: number
  private temperature: number
  private icon: WeatherIcon
  private summary: string

  constructor(
    createdAt: number,
    temperature: number,
    summary: string,
    icon: WeatherIcon
  ) {
    this.icon = icon
    this.summary = summary
    this.createdAt = createdAt
    this.temperature = temperature
  }

  getTemperature(): number {
    return this.temperature
  }

  getSummary(): string {
    return this.summary
  }

  getIcon(): WeatherIcon {
    return this.icon
  }
}
