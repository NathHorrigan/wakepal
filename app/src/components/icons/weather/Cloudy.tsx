import React from 'react'
import Svg, { Path } from 'react-native-svg'
import { Icon } from '../base'
import { colors } from '@utils/theme'

const CloudyIcon: Icon = ({ fill = colors.grey }) => (
  <Svg width={23} height={19} viewBox="0 0 23 19" fill="none">
    <Path
      d="M0 13.64c0-.25.08-.46.253-.63a.86.86 0 01.606-.24h5.474c.748 0 1.384.26 1.91.79a2.6 2.6 0 01.787 1.9c0 .74-.262 1.38-.788 1.9-.525.52-1.161.78-1.909.78-.747 0-1.394-.26-1.919-.79a.802.802 0 01-.232-.6c0-.24.08-.45.232-.6.152-.16.354-.24.606-.24.233 0 .435.08.616.24.202.19.435.29.697.29.263 0 .495-.1.687-.29a.946.946 0 00.293-.7c0-.26-.1-.49-.293-.68a.953.953 0 00-.687-.29H.86a.828.828 0 01-.606-.25.775.775 0 01-.253-.59zm0-3.04a.84.84 0 01.859-.86h10.656c.263 0 .495-.1.687-.29a.956.956 0 000-1.38.953.953 0 00-.687-.29.896.896 0 00-.687.28.982.982 0 01-.646.23.86.86 0 01-.606-.23.81.81 0 01-.233-.6c0-.25.071-.45.233-.61.515-.51 1.161-.76 1.94-.76.747 0 1.393.26 1.918.78.526.52.788 1.15.788 1.88s-.262 1.37-.788 1.89c-.525.52-1.161.78-1.919.78H.86a.827.827 0 01-.606-.24A.74.74 0 010 10.6zm2.172-2.36c0 .08.05.12.161.12h1.455c.08 0 .151-.05.222-.15a3.23 3.23 0 011.06-1.35c.486-.36 1.02-.56 1.607-.6l.535-.08c.131 0 .202-.06.202-.17l.07-.52c.112-1.08.566-1.99 1.385-2.72a4.198 4.198 0 012.889-1.1c1.12 0 2.09.36 2.909 1.09a4.152 4.152 0 011.404 2.73l.07.59c0 .11.061.17.172.17h1.637c.919 0 1.697.32 2.353.96.657.64.98 1.4.98 2.3 0 .89-.323 1.66-.98 2.3-.657.64-1.434.96-2.354.96h-6.98c-.12 0-.191.06-.191.17v1.39c0 .11.06.17.192.17h6.98c.919 0 1.757-.22 2.535-.67a4.963 4.963 0 001.838-1.81c.455-.77.677-1.6.677-2.5 0-.91-.222-1.74-.677-2.5a4.963 4.963 0 00-1.838-1.81 5.053 5.053 0 00-2.535-.67h-.314c-.313-1.33-1.02-2.42-2.12-3.27C14.423.42 13.161 0 11.747 0 10.322 0 9.06.44 7.96 1.32c-1.101.88-1.798 2-2.091 3.37-.869.2-1.637.61-2.303 1.23a4.987 4.987 0 00-1.394 2.21V8.24z"
      fill={fill}
    />
  </Svg>
)

// Exported as WeatherIcon to allow easier importing
export const Cloudy: Icon = React.memo(CloudyIcon)
export default Cloudy
