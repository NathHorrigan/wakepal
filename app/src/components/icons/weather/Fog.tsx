import React from 'react'
import Svg, { Path } from 'react-native-svg'
import { Icon } from '../base'
import { colors } from '@utils/theme'

const FogIcon: Icon = ({ fill = colors.grey }) => (
  <Svg width={24} height={19} viewBox="0 0 24 19" fill="none">
    <Path
      d="M0 15.9c0 .18.063.338.197.465.134.126.3.189.497.189h14.763c.19 0 .347-.063.481-.19a.603.603 0 00.198-.465.62.62 0 00-.197-.473.649.649 0 00-.482-.198H.694a.672.672 0 00-.497.198.631.631 0 00-.197.473zm2.067-2.479c0 .182.063.34.198.458a.63.63 0 00.473.213h14.77c.182 0 .34-.063.466-.197a.668.668 0 00.19-.473.662.662 0 00-.182-.466.624.624 0 00-.465-.19H2.746a.642.642 0 00-.474.19.635.635 0 00-.205.466zm.103-1.893c0 .07.04.102.126.102h1.128c.056 0 .11-.04.166-.126.19-.41.466-.742.836-1.002.371-.26.782-.41 1.231-.442l.426-.055c.087 0 .135-.047.135-.142l.055-.402a3.215 3.215 0 011.08-2.139 3.29 3.29 0 012.257-.86c.853 0 1.602.284 2.241.852a3.17 3.17 0 011.09 2.115l.054.458c0 .086.048.134.15.134h1.27c.506 0 .971.134 1.39.41.418.268.725.631.93 1.081.056.087.103.126.158.126h1.136c.103 0 .143-.055.103-.181l-.158-.434c.6-.742.892-1.61.892-2.612 0-.56-.11-1.089-.324-1.602a4.012 4.012 0 00-.876-1.317 4.07 4.07 0 00-1.317-.884 3.986 3.986 0 00-1.61-.316c-1.215 0-2.225.442-3.014 1.326A4.705 4.705 0 009.61 5.12c-1.104 0-2.09.347-2.95 1.041a4.638 4.638 0 00-1.658 2.66c-1.404.37-2.35 1.246-2.824 2.643-.008.008-.008.031-.008.063zm1.278 6.793c0 .19.071.348.213.474a.61.61 0 00.466.205h14.786c.182 0 .34-.063.474-.197a.649.649 0 00.197-.482.598.598 0 00-.197-.457.696.696 0 00-.474-.182H4.127a.701.701 0 00-.482.182.598.598 0 00-.197.457zM8.632 2.96c0 .19.063.34.182.465l.512.505a.61.61 0 00.458.213.67.67 0 00.473-.197.656.656 0 00.206-.481.697.697 0 00-.198-.497l-.505-.482a.627.627 0 00-.473-.205.62.62 0 00-.473.198.688.688 0 00-.182.48zm4.198 3.456a2.586 2.586 0 011.917-.797c.773 0 1.436.276 1.98.82a2.7 2.7 0 01.821 1.973c0 .513-.126.986-.386 1.42-.75-.75-1.665-1.12-2.738-1.12h-.269c-.228-.931-.67-1.696-1.325-2.296zm1.262-4.127c0 .182.063.34.19.466.126.126.276.19.465.19a.698.698 0 00.497-.19.603.603 0 00.197-.466V.671a.652.652 0 00-.205-.474A.697.697 0 0014.74 0a.614.614 0 00-.465.197.668.668 0 00-.19.474v1.617h.008zm4.332 1.791c0 .174.063.34.19.474.291.284.615.284.97 0l1.128-1.129a.73.73 0 00.19-.505.637.637 0 00-.19-.465.637.637 0 00-.465-.19.642.642 0 00-.474.19l-1.152 1.16a.665.665 0 00-.197.465zm.64 9.311c0 .19.07.347.204.473l.505.513c.127.127.284.19.458.19a.76.76 0 00.481-.19.716.716 0 00.19-.505.585.585 0 00-.19-.442l-.513-.52a.696.696 0 00-.473-.19.62.62 0 00-.473.197.642.642 0 00-.19.474zm1.167-4.979c0 .19.07.347.205.474a.592.592 0 00.466.197h1.617c.182 0 .34-.063.466-.197a.668.668 0 00.189-.474.642.642 0 00-.19-.473.624.624 0 00-.465-.19h-1.617a.62.62 0 00-.474.198.614.614 0 00-.197.465z"
      fill={fill}
    />
  </Svg>
)

// Exported as WeatherIcon to allow easier importing
export const Fog: Icon = React.memo(FogIcon)
export default Fog
