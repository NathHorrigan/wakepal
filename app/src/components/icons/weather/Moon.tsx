import React from 'react'
import Svg, { Path } from 'react-native-svg'
import { Icon } from '../base'
import { colors } from '@utils/theme'

const MoonIcon: Icon = ({ fill = colors.grey }) => (
  <Svg width={17} height={19} viewBox="0 0 17 19" fill="none">
    <Path
      d="M0 8.048C0 6.964.21 5.925.641 4.93a7.968 7.968 0 011.714-2.564A8.307 8.307 0 014.919.64 7.768 7.768 0 018.048 0H9.34c.177.044.265.155.265.31l.056.994c.022.708.165 1.382.442 2.023a5.366 5.366 0 002.742 2.797 5.36 5.36 0 001.99.476l.939.077c.177 0 .265.088.265.254v1.117a7.95 7.95 0 01-.652 3.228h-2.266c.563-.818.917-1.758 1.072-2.797-1.857-.387-3.338-1.183-4.455-2.388a7.241 7.241 0 01-1.934-4.234 5.645 5.645 0 00-2.941.995 6.442 6.442 0 00-2.067 2.3 6.196 6.196 0 00-.73 2.907c0 1.183.31 2.255.918 3.228H.663A8.165 8.165 0 010 8.047zm.276 6.666c0-.288.11-.542.332-.763.199-.2.453-.299.752-.299h3.56c.12 0 .22.023.309.089l2.598 2.454 2.642-2.465a.551.551 0 01.32-.089h3.648c.299 0 .553.1.763.31.21.21.32.464.32.752 0 .298-.11.552-.32.762-.21.21-.464.32-.763.32h-2.962l-3.46 3.14c-.133.1-.266.1-.376 0l-3.405-3.14H1.36c-.299 0-.553-.11-.763-.32a1.002 1.002 0 01-.32-.751z"
      fill={fill}
    />
  </Svg>
)

// Exported as WeatherIcon to allow easier importing
export const Moon: Icon = React.memo(MoonIcon)
export default Moon
