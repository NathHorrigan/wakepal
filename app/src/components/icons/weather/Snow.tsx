import React from 'react'
import Svg, { Path } from 'react-native-svg'
import { Icon } from '../base'
import { colors } from '@utils/theme'

const SnowIcon: Icon = ({ fill = colors.grey }) => (
  <Svg width={30} height={30} viewBox="0 0 30 30" fill="none">
    <Path
      d="M1.58 16.93c0 .86.21 1.67.64 2.41.42.74 1 1.34 1.74 1.79.73.45 1.54.69 2.4.71.11 0 .17-.06.17-.17v-1.33c0-.12-.06-.19-.17-.19-.85-.04-1.58-.38-2.18-1.02-.6-.64-.9-1.37-.9-2.21 0-.82.28-1.54.85-2.16.57-.61 1.26-.97 2.1-1.07l.53-.06c.12 0 .18-.06.18-.19l.08-.51c.11-1.09.56-2 1.36-2.73s1.75-1.09 2.85-1.09c1.09 0 2.04.36 2.85 1.09.82.73 1.28 1.63 1.38 2.7l.07.58c0 .11.06.17.17.17h1.61c.9 0 1.67.32 2.31.96.64.64.96 1.4.96 2.29 0 .84-.3 1.57-.9 2.21-.6.63-1.33.97-2.17 1.02-.12 0-.19.06-.19.19v1.33c0 .11.06.17.19.17 1.33-.04 2.45-.54 3.38-1.5.93-.96 1.39-2.09 1.39-3.41 0-.76-.14-1.43-.43-2.03.75-.93 1.15-2.03 1.15-3.28 0-.94-.23-1.81-.7-2.61-.47-.8-1.11-1.44-1.91-1.91-.8-.47-1.68-.7-2.62-.7-1.54 0-2.83.58-3.87 1.73-.81-.44-1.71-.66-2.69-.66-1.41 0-2.65.44-3.74 1.31-1.09.87-1.78 1.99-2.09 3.34-1.12.28-2.03.86-2.74 1.75-.71.9-1.06 1.92-1.06 3.08zm6.34 4.05c0 .24.08.44.24.61.16.17.35.25.59.25.23 0 .43-.08.59-.25.16-.17.24-.37.24-.61 0-.23-.08-.42-.24-.58a.791.791 0 00-.59-.24c-.23 0-.43.08-.59.24-.16.16-.24.36-.24.58zm0 3.63c0 .21.08.4.24.57.18.16.37.24.58.24.24 0 .43-.08.59-.23.16-.16.23-.35.23-.58 0-.24-.08-.43-.24-.59a.784.784 0 00-.59-.23.84.84 0 00-.59.23c-.14.15-.22.35-.22.59zM9.97 4.68c0 .24.08.44.24.59l.66.66c.16.16.34.25.53.25.21.03.41-.04.61-.22s.3-.39.3-.63-.08-.46-.24-.64l-.64-.61a.73.73 0 00-.58-.25c-.25 0-.46.08-.63.25-.17.16-.25.36-.25.6zM11.1 22.9c0 .22.08.42.24.6.16.16.36.24.58.24.24 0 .44-.08.6-.24.16-.16.25-.36.25-.6 0-.23-.08-.43-.25-.6a.816.816 0 00-.6-.25c-.23 0-.42.08-.58.25-.16.17-.24.37-.24.6zm0-3.6c0 .23.08.42.24.58.16.16.36.24.58.24.24 0 .44-.08.6-.24.17-.16.25-.35.25-.59 0-.23-.08-.43-.25-.59a.847.847 0 00-.6-.24c-.23 0-.42.08-.58.24a.83.83 0 00-.24.6zm0 7.26c0 .22.08.41.24.57.17.17.36.25.58.25.24 0 .44-.08.6-.23.17-.16.25-.35.25-.59s-.08-.44-.25-.6a.816.816 0 00-.6-.25c-.22 0-.41.08-.58.25-.16.17-.24.37-.24.6zm3.22-5.58c0 .24.08.44.24.61.16.17.36.25.59.25.23 0 .43-.08.59-.25.16-.17.24-.37.24-.61 0-.23-.08-.42-.24-.58a.791.791 0 00-.59-.24c-.24 0-.43.08-.59.24-.16.16-.24.36-.24.58zm0 3.63c0 .21.08.4.23.57.18.16.38.24.6.24.24 0 .43-.08.59-.23.16-.16.23-.35.23-.58 0-.24-.08-.43-.24-.59a.784.784 0 00-.59-.23c-.24 0-.44.08-.6.24-.14.15-.22.35-.22.58zm.98-15.55c.69-.66 1.51-.99 2.47-.99.97 0 1.8.35 2.48 1.04.69.69 1.03 1.53 1.03 2.49 0 .62-.17 1.24-.51 1.84-.95-.96-2.11-1.44-3.47-1.44h-.32c-.3-1.17-.86-2.15-1.68-2.94zm1.6-5.22c0 .23.08.43.25.58.17.15.37.23.61.23s.43-.08.59-.23c.16-.16.23-.35.23-.58V1.8c0-.24-.08-.44-.24-.61a.737.737 0 00-.57-.25c-.23 0-.43.09-.6.26-.17.17-.26.37-.26.6v2.04h-.01zm5.52 2.27c0 .23.08.43.25.59.15.16.34.24.56.26.22.02.43-.07.62-.26l1.43-1.43c.18-.18.26-.38.26-.61 0-.24-.09-.44-.26-.61a.816.816 0 00-.6-.25c-.22 0-.41.08-.58.25l-1.43 1.46c-.17.16-.25.36-.25.6zm.8 11.8c0 .25.08.46.24.62l.64.63c.24.16.46.24.64.24.21 0 .39-.09.56-.26.17-.17.25-.38.25-.61 0-.23-.09-.42-.26-.58l-.62-.65a.888.888 0 00-.61-.24c-.23 0-.43.08-.59.25-.17.16-.25.36-.25.6zm1.45-6.31c0 .24.09.43.26.59.17.18.38.27.62.27h2.02c.23 0 .43-.08.6-.25.17-.17.25-.37.25-.61s-.08-.44-.25-.6a.857.857 0 00-.6-.25h-2.02a.86.86 0 00-.62.25.79.79 0 00-.26.6z"
      fill={fill}
    />
  </Svg>
)

// Exported as WeatherIcon to allow easier importing
export const Snow: Icon = React.memo(SnowIcon)
export default Snow
