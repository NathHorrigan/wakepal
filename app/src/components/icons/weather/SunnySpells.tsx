import React from 'react'
import Svg, { Path } from 'react-native-svg'
import { Icon } from '../base'
import { colors } from '@utils/theme'

const SunnySpellsIcon: Icon = ({ fill = colors.grey }) => (
  <Svg width={21} height={19} viewBox="0 0 21 19" fill="none">
    <Path
      d="M0 10.567c0 .233.078.424.242.579.164.155.371.225.622.225h1.46a.835.835 0 00.596-.234.76.76 0 00.251-.57c0-.25-.078-.458-.242-.613a.843.843 0 00-.596-.242H.864a.84.84 0 00-.622.242c-.164.146-.242.354-.242.613zm1.53 5.34c0 .838.302 1.564.915 2.177.614.614 1.33.916 2.169.916h5.927c.838 0 1.555-.302 2.152-.907a3 3 0 00.898-2.186c0-.415-.06-.77-.181-1.063a4.957 4.957 0 001.702-1.832c.415-.76.63-1.58.63-2.454 0-.7-.138-1.373-.414-2.013a5.27 5.27 0 00-1.106-1.65 5.27 5.27 0 00-1.65-1.106 4.989 4.989 0 00-2.005-.415 5.07 5.07 0 00-2.61.7 5.181 5.181 0 00-1.883 1.884 5.108 5.108 0 00-.682 2.6v.294c-.864.493-1.435 1.21-1.711 2.152-.657.207-1.175.57-1.564 1.097a2.973 2.973 0 00-.588 1.806zM2.876 3.672c0 .242.078.44.225.579l1.391 1.46c.406.311.812.311 1.218 0a.814.814 0 00.242-.605.842.842 0 00-.242-.604L4.286 3.076a.896.896 0 00-.605-.25c-.242 0-.44.077-.588.241a.866.866 0 00-.216.605zm.363 12.235c0-.363.112-.674.346-.933.233-.26.527-.406.881-.441l.536-.07c.112 0 .173-.068.173-.198l.077-.484a1.93 1.93 0 01.631-1.244 1.81 1.81 0 011.279-.501c.5 0 .942.164 1.304.501.372.337.588.752.649 1.244l.069.562c.051.13.12.199.207.199h1.14c.372 0 .692.138.968.406.277.268.406.587.406.967 0 .39-.138.718-.406.994a1.308 1.308 0 01-.967.415h-5.92c-.388 0-.716-.139-.975-.415a1.389 1.389 0 01-.398-1.002zm3.82-5.521c.06-.942.422-1.737 1.096-2.385a3.375 3.375 0 012.411-.968c.96 0 1.78.346 2.454 1.028.674.683 1.01 1.521 1.01 2.497 0 .605-.146 1.167-.44 1.685a3.525 3.525 0 01-1.192 1.253 2.669 2.669 0 00-1.754-.63 3.51 3.51 0 00-1.33-1.806 3.61 3.61 0 00-2.144-.674H7.06zm2.669-7.56c0 .233.086.44.25.604.165.164.363.25.597.25.242 0 .44-.086.604-.25a.821.821 0 00.251-.605V.804a.76.76 0 00-.25-.57.841.841 0 00-.605-.234.835.835 0 00-.596.233.76.76 0 00-.251.57v2.022zm5.46 13.159c0 .233.078.432.234.604l1.417 1.4c.363.363.769.363 1.218 0 .156-.147.225-.337.225-.588a.81.81 0 00-.225-.578l-1.46-1.443a.823.823 0 00-.587-.216.801.801 0 00-.82.82zm0-10.896c0 .242.078.45.234.622a.801.801 0 00.588.234.801.801 0 00.587-.234l1.46-1.46c.156-.147.225-.337.225-.579a.789.789 0 00-.25-.596.849.849 0 00-1.193.009l-1.417 1.408a.807.807 0 00-.233.596zm2.23 5.478c0 .233.078.424.242.579.164.155.371.225.622.225h1.46a.835.835 0 00.596-.234.76.76 0 00.25-.57c0-.25-.077-.458-.241-.613a.843.843 0 00-.596-.242h-1.46a.84.84 0 00-.623.242c-.164.146-.25.354-.25.613z"
      fill={fill}
    />
  </Svg>
)

// Exported as WeatherIcon to allow easier importing
export const SunnySpells: Icon = React.memo(SunnySpellsIcon)
export default SunnySpells
