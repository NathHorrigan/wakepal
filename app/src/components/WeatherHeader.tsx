import React, { SFC } from 'react'
import styled from 'styled-components/native'

import WeatherIcons from '@components/icons/weather'
import { WeatherResponse } from '@api/weather/WeatherResponse'
import { colors, fonts } from '@utils/theme'

interface WeatherHeaderProps {
  name: string
  dateLabel: string
  weather: WeatherResponse
}

const WeatherHeader: SFC<WeatherHeaderProps> = ({
  name,
  dateLabel,
  weather,
}) => {
  const WeatherIcon = WeatherIcons[weather.getIcon()]
  return (
    <HeaderContainer>
      <LeftColumn>
        <WelcomeText>
          {getGreeting()}, {name}
        </WelcomeText>
        <DateText>{dateLabel}</DateText>
      </LeftColumn>
      <RightColumn>
        <WeatherIcon />
        <WeatherText>
          <HighlightText>{weather.getTemperature()}Â° </HighlightText>
          {weather.getSummary()}
        </WeatherText>
      </RightColumn>
    </HeaderContainer>
  )
}

function getGreeting(): string {
  const hourOfDay = new Date().getHours()
  if (hourOfDay < 12) {
    return 'Morning'
  }

  if (hourOfDay < 15) {
    return 'Afternoon'
  }

  if (hourOfDay < 18) {
    return 'Evening'
  }

  if (hourOfDay < 21) {
    return 'Goodnight'
  }

  return 'Hello'
}

const HeaderContainer = styled.View`
  width: 100%;
  height: 50px;
  padding: 0 20px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`

const LeftColumn = styled.View`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`

const RightColumn = styled(LeftColumn)`
  align-items: flex-end;
  justify-content: flex-end;
  top: -1px;
`

const DateText = styled.Text`
  color: black;
  font-family: ${fonts.semiBold};
  font-size: 25px;
`

const WeatherText = styled(DateText)`
  font-size: 15px;
  text-transform: capitalize;
`

const HighlightText = styled.Text`
  color: ${colors.coral};
`

const WelcomeText = styled.Text`
  color: ${colors.grey};
  font-family: ${fonts.medium};
  font-size: 16px;
`

export default WeatherHeader
