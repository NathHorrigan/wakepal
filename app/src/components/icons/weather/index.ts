import Sunny from './Sunny'
import PartlySunny from './PartlySunny'
import Cloudy from './Cloudy'
import Gloomy from './Gloomy'
import SunnySpells from './SunnySpells'
import Lightning from './Lightning'
import Snow from './Snow'
import Fog from './Fog'
import Moon from './Moon'

export default {
  Sunny: Sunny,
  PartlySunny: PartlySunny,
  Cloudy: Cloudy,
  Gloomy: Gloomy,
  SunnySpells: SunnySpells,
  Lightning: Lightning,
  Snow: Snow,
  Fog: Fog,
  Moon: Moon,
}
