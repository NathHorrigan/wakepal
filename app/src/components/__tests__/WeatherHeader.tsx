import React from 'react'
import WeatherHeader from '../WeatherHeader'
import renderer from 'react-test-renderer'
import { WeatherResponse, WeatherIcon } from '@api/weather/WeatherResponse'

export const exampleWeather = new WeatherResponse(
  Date.now,
  6,
  'cloudy',
  WeatherIcon.PartlySunny
)

it('renders correctly', () => {
  const tree = renderer
    .create(
      <WeatherHeader name="Nathan" dateLabel="Today" weather={exampleWeather} />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
