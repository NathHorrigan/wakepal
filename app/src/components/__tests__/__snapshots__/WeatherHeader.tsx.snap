// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "space-between",
        "paddingBottom": 0,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 0,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#9E9E9E",
            "fontFamily": "SFProRounded-Medium",
            "fontSize": 16,
          },
        ]
      }
    >
      Afternoon
      , 
      Nathan
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontFamily": "SFProRounded-SemiBold",
            "fontSize": 25,
          },
        ]
      }
    >
      Today
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-end",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={19}
      bbWidth={25}
      fill="none"
      focusable={false}
      height={19}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 19,
            "width": 25,
          },
        ]
      }
      vbHeight={19}
      vbWidth={25}
      width={25}
    >
      <RNSVGGroup
        fill={null}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M0 14.509c0 .816.2 1.57.599 2.259.4.69.944 1.234 1.633 1.633.69.4 1.434.599 2.241.599H14.3c.807 0 1.56-.2 2.25-.599.69-.4 1.243-.944 1.642-1.633.4-.69.608-1.443.608-2.26 0-.598-.127-1.206-.38-1.814.689-.835 1.034-1.842 1.034-2.994 0-.645-.127-1.262-.372-1.851a4.635 4.635 0 00-1.017-1.516 4.635 4.635 0 00-1.515-1.016 4.622 4.622 0 00-1.85-.372c-1.344 0-2.514.526-3.521 1.579-.7-.4-1.516-.599-2.45-.599-1.28 0-2.405.4-3.385 1.189a5.262 5.262 0 00-1.887 3.04 4.308 4.308 0 00-2.486 1.569A4.375 4.375 0 000 14.509zm1.552 0c0-.763.254-1.416.762-1.97.508-.553 1.143-.87 1.905-.961l.454-.027c.109 0 .172-.055.172-.164l.064-.49c.127-.98.553-1.805 1.28-2.459a3.663 3.663 0 012.549-.989c.998 0 1.869.336 2.604.998a3.62 3.62 0 011.243 2.46l.063.526c.019.1.082.154.191.154h1.46c.8 0 1.489.29 2.07.871.58.58.87 1.261.87 2.06a2.89 2.89 0 01-.861 2.105c-.572.58-1.27.87-2.069.87H4.473c-.798 0-1.479-.29-2.06-.88-.571-.59-.861-1.288-.861-2.104zM7.63 3.375c0 .218.072.409.218.572l.598.58c.227.173.418.246.581.228.19 0 .354-.082.5-.236a.796.796 0 00.217-.563c0-.218-.082-.4-.236-.535l-.535-.6a.806.806 0 00-.554-.217.766.766 0 00-.562.227.713.713 0 00-.227.544zm4.845 4.02a3.085 3.085 0 012.223-.907c.898 0 1.66.308 2.287.934.626.626.943 1.38.943 2.278 0 .562-.154 1.125-.462 1.67-.88-.872-1.933-1.307-3.167-1.307h-.29c-.227-.99-.735-1.879-1.534-2.668zm1.48-4.79c0 .235.072.416.208.562.136.145.317.208.535.208a.782.782 0 00.563-.208c.145-.146.208-.327.208-.563V.744a.718.718 0 00-.217-.535C15.107.064 14.926 0 14.699 0a.747.747 0 00-.535.209c-.145.136-.209.317-.209.535v1.86zm5.008 2.077c0 .236.063.417.2.563.19.145.38.217.562.217.163 0 .345-.072.535-.217l1.298-1.298a.84.84 0 00.217-.58c0-.218-.072-.4-.217-.545a.732.732 0 00-.536-.218.684.684 0 00-.526.218l-1.334 1.297a.872.872 0 00-.2.563zm.716 10.743c0 .218.073.408.227.572l.59.571a.684.684 0 00.526.218c.218 0 .4-.073.545-.227a.78.78 0 00.218-.562c0-.2-.073-.381-.218-.527l-.59-.59a.706.706 0 00-.517-.217c-.218 0-.4.072-.545.218a.72.72 0 00-.236.544zM21.014 9.7c0 .208.082.38.236.526a.76.76 0 00.554.218h1.85a.74.74 0 00.527-.209c.145-.136.209-.318.209-.535 0-.218-.073-.4-.209-.545a.706.706 0 00-.526-.227h-1.851c-.218 0-.4.073-.554.227a.716.716 0 00-.236.545z"
          fill={4288585374}
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontFamily": "SFProRounded-SemiBold",
            "fontSize": 20,
            "textTransform": "capitalize",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#F86A6A",
            },
          ]
        }
      >
        6
        Â° 
      </Text>
      cloudy
    </Text>
  </View>
</View>
`;
